sympy/geometry/line.py

```diff
@@ -1070,6 +1070,40 @@ def contains(self, other):
             return Point.is_collinear(self.p1, self.p2, other.p1, other.p2)
         return False
 
+    def coordinate(self, other):
+        """
+        Finds the parameter ``t`` for which
+        ``other == t * self.p1 + (1 - t) self.p2`` is True.
+
+        Raises
+        ======
+
+        ValueError
+            When ``other`` is not contained in the line.
+
+        NotImplementedError
+            When ``other`` is not a ``Point``.
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Line
+        >>> p1, p2, p3 = Point(0, 1), Point(2, 3), Point(4, 5)
+        >>> l = Line(p1, p2)
+        >>> l.coordinate(p3)
+        2
+        """
+        if isinstance(other, Point):
```

Comment:
insert `other = Point(other)`

Reply:
Actually, perhaps better to say `other = Point._normalize_dimension(Point(other), dim=self.p1.ambient_dimension)` if this should work for a 3-D line.

Reply:
Hey ! @smichr , 
`other = Point._normalize_dimension(Point(other),dim=self.p1.ambient_dimension) ` is giving a `Value error` .I tried many things but was unable to remove the error. It would be great if you can tell me where I am going wrong . The error that I am getting is : 
```
  File "/home/yash_jain/Desktop/env/sympy/sympy/geometry/tests/test_line.py", line 293, in test_coordinate
    assert l.coordinate(p3) == Rational(3, 2)
  File "/home/yash_jain/Desktop/env/sympy/sympy/geometry/line.py", line 1098, in coordinate
    if Point.is_collinear(self.p1, self.p2, other):
  File "/home/yash_jain/Desktop/env/sympy/sympy/geometry/point.py", line 538, in is_collinear
    points = Point._normalize_dimension(*[Point(i) for i in points])
  File "/home/yash_jain/Desktop/env/sympy/sympy/geometry/point.py", line 538, in <listcomp>
    points = Point._normalize_dimension(*[Point(i) for i in points])
  File "/home/yash_jain/Desktop/env/sympy/sympy/geometry/point.py", line 141, in __new__
    keyword `dim` > 1.'''))
ValueError: 
Point requires 2 or more coordinates or keyword `dim` > 1.
```

Reply:
How about `other = self.p1._normalize_dimension(Point(other))`?

Reply:
And that should be the first line of the function.